src/components/NewClusterModal.js

```diff
@@ -149,6 +150,66 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
     ]))
   }
 
+  updateCluster(isStopRequired = false) {
+    const { currentCluster, onSuccess } = this.props
+    const { googleProject, clusterName } = currentCluster
+
+    if (isStopRequired) {
+      notify('success', 'To be updated, your runtime will now stop, and then start.', { timeout: 3000 })
+    }
+
+    return onSuccess(
+      Ajax().Clusters.cluster(googleProject, clusterName).update({
+        machineConfig: this.getMachineConfig()
+      }),
+      isStopRequired ? 5000 : 0)
+  }
+
+  //determines whether the changes are applicable for a call to the leo patch endpoint
+  //see this for a diagram of the conditional this implements https://drive.google.com/file/d/1mtFFecpQTkGYWSgPlaHksYaIudWHa0dY/view
```

Comment:
Should we have documentation accessible to users to show which updates require a full restart of their cluster and which don't?

Reply:
cc @sharmatime what do you think? I can take these documents to the appropriate person  and answer any questions they may have if you think we should and point me in the right direction.